kong/plugins/basic-auth/access.lua

```diff
@@ -158,21 +158,22 @@ end
 
 local function do_authentication(conf)
   local www_authenticate = "Basic realm=\"" .. conf.realm .. "\""
+  local ctx = {}
```

Comment:
Use a local `ctx` here which is cheaper than`ngx.ctx`, and the cache is only fresh in the local `ctx` lifetime, not in the request lifetime, so we don't need `ngx.ctx`